students/TacSPx/lesson03/mailroom.py

```diff
@@ -0,0 +1,156 @@
+            for donor in donor_sort:
+                donor_lst = (row(Name=donor[0]))
+                print(donor_lst)
+            input("Press Enter to Continue")
+            print()
+            continue
+
+        if (donor_name.lower() != 'list'):
+            donor_amount = int(input("Enter Amount of Donation: $"))
+            # if name is in list append dollar amount to donors bucket
+            for i in donor_info:
+                if i[0] == donor_name:
+                    i[1].append(donor_amount)
+                    print(f"'{donor_name}' is in registry added new donation amount!")
+                    break
+                # if name not in list add new name to donor info list
+                else:
+                    donor_info.append((donor_name, [donor_amount]))
+                    print(f"New donor name '{donor_name}' added to registry!")
+                    # print(donor_info) - for testing
+                    break
+            input("\nPress Enter for Thank You letter:\n")
+            print(f'''Dear {donor_name},
+
+Thank you for your donation of '${donor_amount}' to XYZ Nonprofit for children! It really makes a huge impact for the children 
+in our community.
+
+Those three hours after the end of the school day can make a crucial difference in kids’ lives.
+Thanks to you, kids have a safe place to go after school. Instead of going home alone while their families are at work,
+our kids are learning to play sports, create art, and improving their grades at our Homework Help Center.  All while
+forming friendships with peers and relationships with adult mentors and tutors.
+
+Thank you again {donor_name}, for your ongoing support of our kids!
+
+Sincerely,
+
+XYZ Nonprofit Agency Director
+''')
+            input("Press Enter to Continue")
+            break
+
+###############################################
+#   #2 Create a Report
+###############################################
+def report():
+    ''' Report: creates a tabular summary report of donor transactions'''
+    while (True):
+    # title
+        title = "*******************  XYZ Nonprofit Charity  ********************\n" \
+                        "________________________________________________________________"
+        heading_name = ("| Donor Name:", "| Totals $:", "| #-of-Gifts:", "| Avg Gift $: ", "|")
+        header = f"{heading_name[0].upper():<20s} {heading_name[1].upper():<13s}" \
+                    f" {heading_name[2].upper():<6s} {heading_name[3].upper():<14s}|"
+        print(title)
+        print(header)
+
+        #define format row
+        row = "| {Name:<18s} | ${Total:<10.2f} | {NumbGifts:^11d} | ${AverageGifts:<10.2f} |".format
+
+        for i in donor_info:
+            print(row(Name=i[0], Total=(sum(i[1])), NumbGifts=len(i[1]), AverageGifts=sum(i[1], 0) / len(i[1])))
+        print("________________________________________________________________")
+        input("Press Enter to Continue")
+        break
```

Comment:
Missing the sort function - donors should be listed by total donated, in descending order.